samplesheets/views_api.py

```diff
@@ -421,3 +422,49 @@ def get(self, request, **kwargs):
             except Exception as ex:
                 return Response(str(ex), status=500)
         return Response(ret, status=200)
+
+
+class ProjectIrodsFileListAPIView(
+    SODARAPIBaseProjectMixin, SODARAPIBaseMixin, APIView
+):
+    """
+    Return a list of files in the project's sample data repository.
+
+    **URL:** ``/samplesheets/api/irods/files/{Project.sodar_uuid}``
+
+    **Methods:** ``GET``
+
+    **Parameters:**
+
+    - ``project``: Project SODAR UUID
+
+    **Returns:**
+
+    - ``irods_data``: List of iRODS data objects
+    """
+
+    http_method_names = ['get']
+    permission_required = 'samplesheets.view_sheet'
+
+    def get(self, request, *args, **kwargs):
+        if not settings.ENABLE_IRODS:
+            raise APIException('iRODS not enabled')
+        irods_backend = get_backend_api('omics_irods')
+        if not irods_backend:
+            raise APIException('iRODS backend not enabled')
+        project = self.get_project(request=request, kwargs=kwargs)
+        self.check_object_permissions(request, project)
+        path = request.query_params.get('path')
+        if not path:
+            path = irods_backend.get_sample_path(project)
+        else:
+            path = os.path.join(irods_backend.get_sample_path(project), path)
+        try:
+            with irods_backend.get_session() as irods:
+                irods_data = irods_backend.get_objects(irods, path)
+        except Exception as ex:
+            return Response(
+                {'detail': 'Unable to connect to iRODS: {}'.format(ex)},
```

Comment:
We actually can't know that any exception returned by `get_objects()` would mean a connection to iRODS can not be made. There may be other things going wrong. Instead, report this back as e.g. "Exception querying iRODS objects".